exercise1.py

```diff
@@ -0,0 +1,27 @@
+import re
+class SerialAverage:
+	
+	def __init__(self,inputString : str) -> None:
+		self.inputString = inputString
+
+
+	def validateString(self): 
+		validate = re.compile(r'^[0-9]{3}-[0-9]{2}.[0-9]{2}-[0-9]{2}.[0-9]{2}')
+		if bool(re.match(validate, self.inputString)) == False:
+			raise TypeError
+	
+	def getAverage(self):
+		try:
+			self.validateString()
+			resStr = self.inputString.split('-')
+			avg = resStr[0]+"-" 	
+			avg += str(format((((float(resStr[len(resStr)-2])+float(resStr[len(resStr)-1]))/2)), '.2f'))
+			return avg
+		except TypeError:
+			return "Input recieved is of invalid format"
+
```

Comment:
not a good practice to return a string, instead use raise

Reply:
Done